tests/client/amo/components/TestAddonCompatibilityError.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import {
+  renderIntoDocument,
+  findRenderedComponentWithType,
+} from 'react-addons-test-utils';
+import { findDOMNode } from 'react-dom';
+import { Provider } from 'react-redux';
+
+import AddonCompatibilityError from 'amo/components/AddonCompatibilityError';
+import createStore from 'amo/store';
+import { signedInApiState } from 'tests/client/amo/helpers';
+import { getFakeI18nInst } from 'tests/client/helpers';
+import I18nProvider from 'core/i18n/Provider';
+
+
+describe('AddonCompatibilityError', () => {
+  function render({ ...props }) {
+    const api = {
+      ...signedInApiState,
+      lang: props.lang || signedInApiState.lang,
+      userAgentInfo: props.userAgentInfo,
+    };
+
+    return findRenderedComponentWithType(renderIntoDocument(
+      <Provider store={createStore({ api })}>
+        <I18nProvider i18n={getFakeI18nInst()}>
+          <AddonCompatibilityError {...props} />
+        </I18nProvider>
+      </Provider>
+    ), AddonCompatibilityError);
+  }
+
+  it('logs a warning and renders nothing if compatible', () => {
+    const fakeLog = { warn: sinon.stub() };
+    const userAgentInfo = {
+      browser: { name: 'Firefox' },
+      os: { name: 'Plan 9' },
+    };
+    const root = render({ log: fakeLog, userAgentInfo });
+    const rootNode = findDOMNode(root);
+
+    assert.include(rootNode.textContent, '');
+    assert.include(fakeLog.warn.firstCall.args[0],
+      'no reason to mark the add-on as incompatible');
+    assert.deepEqual(fakeLog.warn.firstCall.args[1], userAgentInfo);
+  });
+
+  it('renders a notice for non-Firefox browsers', () => {
+    const root = render({
+      lang: 'es',
+      userAgentInfo: { browser: { name: 'Chrome' }, os: {} } });
+    const rootNode = findDOMNode(root);
+
+    // This is localised to the current locale, which is en-US in our
+    // signedInApiState helper.
```

Comment:
You can delete this comment. And, yeah, I like the explicit approach of passing in `lang` better.